tests/integration/test_time_series.py

```diff
@@ -5,7 +5,7 @@
 import pytest
 import time
 
-from rat import ops
+from rat import ast
```

Comment:
The time series tests aren't passing. I suspect something is broken with recursive assignment.

Reply:
Yeah this is bothering me alot lol. I manually checked the generated subscript indices and they match the original compiler. Still poking around to see where it's coming from.

Reply:
Woof, probably some really fringe assumption I'm making with sorting or something lol.

Reply:
I found a bug with setting pad_needed which resolved an error; two more to go!

Reply:
I found the culprit! It was calculating `first_in_group_indicator` based on the joined dataframe instead of the variable's base_df.

Reply:
These integration tests are absolutely awesome <3

Reply:
Hell yeah hell yeah good sleuthing!

I've been really happy with what integration tests we have too. In that spirit, I'm pretty happy not going to a finer level of tests yet. So many of the details are changing so rapidly that I don't think going lower than this would be great.